src/cbc_sdk/audit_remediation/base.py

```diff
@@ -588,8 +633,36 @@ def submit(self):
 
         return self._doc_class(self._cb, initial_data=resp.json())
 
+    def _init_async_query(self):
+        """
+        Initialize an async query and return a context for running in the background. Optional.
+
+        Returns:
+            Run: Context for running in the background.
+        """
+        return self.submit()
+
+    def _run_async_query(self, context):
+        """
+        Executed in the background to run an asynchronous query. Must be implemented in any inheriting classes.
 
-class RunHistoryQuery(BaseQuery, QueryBuilderSupportMixin, IterableQueryMixin, CriteriaBuilderSupportMixin):
+        Args:
+            context (object): The context returned by _init_async_query. May be None.
+
+        Returns:
+            Any: Result of the async query, which is then returned by the future.
+        """
+        while context.status == 'ACTIVE':
```

Comment:
Wouldn't those states `ACTIVE, COMPLETE etc..` be better in an `enum`?

Reply:
To turn them into an enum now would break existing code, which we prefer not to do unless there's no way around it.